metrics/exporter.go

```diff
@@ -244,6 +260,10 @@ func setCurMetricsConfig(c *metricsConfig) {
 // Return value indicates whether the exporter is flushable or not.
 func FlushExporter() bool {
 	e := getCurMetricsExporter()
+	return flushExporterUnlocked(e)
+}
+
+func flushExporterUnlocked(e view.Exporter) bool {
```

Comment:
Maybe `flushGivenExporter`. Since you pass the exporter as parameter, we don't need to care about whether it is locked or not.